magento2/usr/local/share/magento2/magento_functions.sh

```diff
@@ -34,28 +34,44 @@ function do_magento_create_web_writable_directories() {
 }
 
 function do_magento_frontend_build() {
-  if [ -d "$FRONTEND_INSTALL_DIRECTORY" ]; then
-    mkdir -p pub/static/frontend/
+  do_magento_frontend_build_install
+  do_magento_frontend_build_run
+}
+
+function do_magento_frontend_build_install() {
+  if [ ! -d "$FRONTEND_INSTALL_DIRECTORY" ]; then
+    return
+  fi
 
+  mkdir -p pub/static/frontend/
   if [ -d "pub/static/frontend/" ] && [ "$IS_CHOWN_FORBIDDEN" != 'true' ]; then
-      chown -R "${CODE_OWNER}:${CODE_GROUP}" pub/static/frontend/
-    fi
+    chown -R "${CODE_OWNER}:${CODE_GROUP}" pub/static/frontend/
+  fi
 
-    if [ ! -d "$FRONTEND_INSTALL_DIRECTORY/node_modules" ]; then
-      as_code_owner "npm install" "$FRONTEND_INSTALL_DIRECTORY"
-    fi
-    if [ -z "$GULP_BUILD_THEME_NAME" ]; then
-      as_code_owner "gulp $FRONTEND_BUILD_ACTION" "$FRONTEND_BUILD_DIRECTORY"
-    else
-      as_code_owner "gulp $FRONTEND_BUILD_ACTION --theme='$GULP_BUILD_THEME_NAME'" "$FRONTEND_BUILD_DIRECTORY"
-    fi
+  as_code_owner "npm install" "$FRONTEND_INSTALL_DIRECTORY"
```

Comment:
Is npm standard in m2 or just the way we do things? should we check for the presence of package.json?

Reply:
Good point, I've made it check now